api/src/main/java/com/google/android/apps/dashclock/api/host/DashClockHost.java

```diff
@@ -202,12 +208,14 @@ public Intent getEnableForceWorldReadabilityIntent() {
      */
     public List<ExtensionListing> getAvailableExtensions(boolean onlyWorldReadableExtensions) {
         if (!onlyWorldReadableExtensions) {
-            return mAvailableExtensions;
+            return new ArrayList<>(mAvailableExtensions);
         }
-        List<ExtensionListing> eis = new ArrayList<>();
-        for (ExtensionListing ei : mAvailableExtensions) {
-            if (ei.worldReadable()) {
-                eis.add(ei);
+        List<ExtensionListing> eis = new ArrayList<>(mAvailableExtensions);
```

Comment:
I wanted to avoid a concurrent exception on mAvailableExtensions iteration, so I add all the extensions refs to a local array and do a safe iteration over it. Since the all the extensions are in the array right now, a remove operation is needed instead of an add operation.